kanin/src/extract/req_id.rs

```diff
@@ -0,0 +1,68 @@
+//! Req id ...
+
+use core::fmt;
+use std::convert::Infallible;
+
+use async_trait::async_trait;
+use lapin::types::{AMQPValue, LongString};
+use uuid::Uuid;
+
+use crate::{Extract, Request};
+
+/// Request id.
+pub struct ReqId(AMQPValue);
```

Comment:
```suggestion
/// Request IDs allow concurrent logs to be associated with a unique request. It can also enable requests
/// to be traced between different services by propagating the request IDs when calling other services.
/// This type implements [`Extract`], so it can be used in handlers.
pub struct ReqId(pub AMQPValue);
```